tests/syslog/syslog_utils.py

```diff
@@ -121,26 +129,30 @@ def capture_syslog_packets(dut, tcpdump_cmd, logger_flags='', logger_msg=''):
         tcpdump_cmd (str): tcpdump cmd
     Return: filepath
     """
-    logging.info("Start tcpdump: {}".format(tcpdump_cmd))
+    logging.info(f"Start tcpdump: {tcpdump_cmd}")
 
-    pcap_file_full_path = tcpdump_cmd.split("-w")[-1].strip()
-    dut.shell("sudo rm -f {}".format(pcap_file_full_path))
+    dut_pcap_filepath = tcpdump_cmd.split("-w")[-1].strip()
+    dut.shell("sudo rm -f {}".format(dut_pcap_filepath))
     tcpdump_task, tcpdump_result = dut.shell(tcpdump_cmd, module_async=True)
     # wait for starting tcpdump
-    time.sleep(TSHARK_START_TIME)
+    time.sleep(syslogUtilsConst.TSHARK_START_TIME)
 
     logging.debug("Generating log message from DUT")
     # Generate syslog msgs from the DUT
-    logger_info_msg_count = 20
     default_priority = '--priority INFO'
-    random_msg = ''.join(random.choice(string.ascii_letters) for _ in range(logger_info_msg_count))
-    for i in range(logger_info_msg_count):
-        dut.shell("logger {flags} ....{msg}".format(flags=default_priority+''+logger_flags, msg=random_msg+logger_msg))
-        time.sleep(0.2)
+    for flag, msg in logging_data:
+        for i in range(syslogUtilsConst.PACKETS_NUM):
+            dut.shell(f"logger {default_priority} {flag} {msg} {i + 1}")
```

Comment:
This command run in host.

However, the config causes this issue are:
'''
if re_match($programname, "bgp[0-9]*#(frr|zebra|staticd|watchfrr)") then {
    /var/log/frr/zebra.log
    stop
}
'''

Which means when reproduce the issue, the log send inside bgp docker container, then the 'programname' part can match.

I warry about this test case can't prevent regression. can you test with an image have this issue to make sure this test can catch issue?

Reply:
This is good suggestion: test with an image have this issue to make sure this test can catch issue